File: AUGUST 23/01 August/DFS of Graph gfg.cpp
    void dfs(int root, vector<int> adj[], vector<bool> &visited, vector<int> &traversal){
        visited[root] = true;
        traversal.push_back(root);

        for(int next:adj[root])
        {
            if(!visited[next])
                dfs(next, adj, visited, traversal);
        }
    }
    vector<int> dfsOfGraph(int V, vector<int> adj[]) {
        // Code here
        vector<int> traversal;
        vector<bool> visited(V, false);
        dfs(0, adj, visited, traversal);
        return traversal;
    }

File: APRIL 2025/01 April/DFS of Graph gfg.java

class Solution {
    // Function to return a list containing the DFS traversal of the graph.
    public void traverse(ArrayList<ArrayList<Integer>> adj, ArrayList<Integer> traversal, int node, boolean[] visited) {
        visited[node] = true;
        traversal.add(node);
        for(Integer next:adj.get(node))
        {
            if(!visited[next])
                traverse(adj, traversal, next, visited);
        }
    }
    public ArrayList<Integer> dfs(ArrayList<ArrayList<Integer>> adj) {
        // Code here
        ArrayList<Integer> traversal = new ArrayList<Integer>();
        int N = adj.size();
        boolean[] visited = new boolean[N];
        for(int i=0; i<N; i++) {
            if(!visited[i])
                traverse(adj, traversal, 0, visited);
        }
        return traversal;
    }
}
